cpp/src/arrow/flight/sql/example/sqlite_type_info.cc

```diff
@@ -0,0 +1,116 @@
+  auto literal_suffix = ArrayFromJSON(
+      utf8(),
+      R"([null, null, null, null, null, "'", "'", "'", null, null, null, null, null, "'", "'", "'", "'"])");
+  auto create_params = ArrayFromJSON(
+      list(field("item", utf8(), false)),
+      R"([[], [], [], [], [], ["length"], ["length"], ["length"], [], [], [], [], [], ["length"], [], [], []])");
+  auto nullable =
+      ArrayFromJSON(int32(), R"([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1])");
+  auto case_sensitive =
+      ArrayFromJSON(boolean(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto serachable =
+      ArrayFromJSON(int32(), R"([3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3])");
+  auto unsigned_attribute =
+      ArrayFromJSON(boolean(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto fixed_prec_scale =
+      ArrayFromJSON(boolean(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto auto_unique_value =
+      ArrayFromJSON(boolean(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto local_type_name =
+      ArrayFromJSON(utf8(), R"(["bit", "tinyint", "bigint", "longvarbinary",
+                          "varbinary", "text", "longvarchar", "char",
+                          "integer", "smallint", "float", "double",
+                          "numeric", "varchar", "date", "time",
+                          "timestamp"])");
+  auto minimal_scale =
+      ArrayFromJSON(int32(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto maximum_scale =
+      ArrayFromJSON(int32(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto sql_data_type =
+      ArrayFromJSON(int32(), R"([-7,-6,-5,-4,-3,-1,-1,1,4,5,6,8,8,12,91,92,93])");
+  auto sql_datetime_sub =
+      ArrayFromJSON(int32(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto num_prec_radix =
+      ArrayFromJSON(int32(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+  auto interval_precision =
+      ArrayFromJSON(int32(), R"([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])");
+
+  return RecordBatch::Make(
+      SqlSchema::GetTypeInfoSchema(), 17,
+      {type_name_array, data_type, column_size, literal_prefix, literal_suffix,
+       create_params, nullable, case_sensitive, serachable, unsigned_attribute,
+       fixed_prec_scale, auto_unique_value, local_type_name, minimal_scale, maximum_scale,
+       sql_data_type, sql_datetime_sub, num_prec_radix, interval_precision});
+}
+
+std::shared_ptr<RecordBatch> DoGetTypeInfoResult(int data_type_filter) {
+  auto record_batch = DoGetTypeInfoResult();
+
+  std::vector<int16_t> data_type_vector{-7, -6, -5, -4, -3, -1, -1, 1, 4,
+                                        5,  6,  8,  8,  12, 91, 92, 93};
+
+  // Checking if the data_type is in the vector with the sqlite3 data types
+  auto it = std::find(data_type_vector.begin(), data_type_vector.end(), data_type_filter);
```

Comment:
Can just use `std::equal_range`, e.g. (untested):
```c++
auto pair = std::equal_range(data_type_vector.begin(), data_type_vector.end(), data_type_filter);
return record_batch->Slice(
    /*offset=*/ pair.first - data_type_vector.begin(), /*length=*/ pair.second - pair.first);
```

Reply:
Thanks, this is more simple and it works